src/measure.rs

```diff
@@ -116,92 +154,153 @@ async fn create_provider(endpoint: &Endpoint, api: &Api) -> Result<Provider> {
 
-    let (iterations, skipped) = measurement_loop(&mut measurement_context).await?;
+        let start_time = SystemTime::now();
 
-    measurement_context.progress.finish();
+        let progress = if measurement_config.run_forever {
+            ProgressBar::new_spinner().with_style(
+                // TODO: Add average latency etc...
+                ProgressStyle::with_template("[{elapsed_precise}] {wide_msg} {pos:>7} seconds")
+                    .unwrap(),
+            )
+        } else {
+            ProgressBar::new(measurement_config.run_seconds).with_style(
+                ProgressStyle::with_template(
+                    "[{elapsed_precise}] {msg} [{wide_bar}] {pos:>7}/{len:7} seconds",
+                )
+                .unwrap()
+                .progress_chars("=> "),
+            )
+        };
+        multi_progress_bar.add(progress.clone());
+
+        let group_name = group.group_name.clone();
+
+        let mut measurement_context = MeasurementContext {
+            measurement_config,
+            group_name: group_name.clone(),
+            shutdown_handler,
+            provider,
+            signals,
+            subscriber,
+            progress,
+            hist,
+            running_hist,
+        };
+
+        tasks.spawn(async move {
+            let (iterations_executed, iterations_skipped) =
+                measurement_loop(&mut measurement_context).await.unwrap();
+
+            measurement_context.progress.finish();
+
+            Ok(MeasurementResult {
+                measurement_context,
+                iterations_executed,
+                iterations_skipped,
+                start_time,
+            })
+        });
+    }
 
-    write_output(
-        start_time,
-        measurement_context.measurement_config.interval,
-        iterations,
-        measurement_context.signals.len() as u64,
-        skipped,
-        measurement_context.hist,
-        api,
-    )
-    .await?;
+    let mut measurements_results = Vec::<MeasurementResult>::new();
+    while let Some(received) = tasks.join_next().await {
+        match received {
+            Ok(Ok(measurement_result)) => {
+                // Whenever the first measurement result is received it should stop the execution of the remaning running groups.
+                measurement_result
+                    .measurement_context
+                    .shutdown_handler
+                    .write()
+                    .await
+                    .state
+                    .running
+                    .store(false, Ordering::SeqCst);
+
+                measurements_results.push(measurement_result);
+            }
+            Ok(Err(err)) => {
+                error!("{}", err.to_string());
+                break;
+            }
+            Err(err) => {
+                error!("{}", err.to_string());
+                break;
+            }
+        }
+    }
 
+    print!("\n\nSummary:");
```

Comment:
I think it's better if all output goes through `Term::stdout()` as done in `fn write_output(...)`, instead of using `print!`

Reply:
Done